tests/testthat/test-SurveyFit.R

```diff
@@ -470,3 +470,26 @@ test_that("force factor works appropriately",{
   x <- c(1,1, NA)
   expect_equal(force_factor(x), c(1,1, NA))
 })
+
+
+test_that("custom prediction function works", {
+  myfit <- function(data, formula, ...) {
+    rstanarm::stan_glm(formula = formula, data = data, ...)
+  }
+  mypred <- function(mod, ps, ...) {
+    t(suppressMessages(rstanarm::posterior_linpred(
+      mod,
+      newdata = ps,
+      transform = TRUE
+    )))
+  }
```

Comment:
I'm using rstanarm functions here but wrapping them in a separate function to approximate some custom user function

Reply:
Cool, nice idea!